pkg/auth/auth.go

```diff
@@ -147,22 +148,69 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 		return nil
 	}
 	if unauthorized, ok := err.(docker.ErrUnauthorizedForCredentials); ok {
-		logrus.Debugf("error logging into %q: %v", server, unauthorized)
-		return errors.Errorf("error logging into %q: invalid username/password", server)
+		logrus.Debugf("error logging into %q: %v", key, unauthorized)
+		return errors.Errorf("error logging into %q: invalid username/password", key)
 	}
-	return errors.Wrapf(err, "authenticating creds for %q", server)
+	return errors.Wrapf(err, "authenticating creds for %q", key)
 }
 
-// getRegistryName scrubs and parses the input to get the server name
-func getRegistryName(server string) string {
+// getCredentials returns the auth config for the provided key as well as its registry.
+func getCredentials(systemContext *types.SystemContext, key string, acceptRepositories bool) (auth types.DockerAuthConfig, registry string, err error) {
+	// Apply a more restrictive input validation if acceptRepositories is
+	// allowed
+	if acceptRepositories {
+		if strings.HasPrefix(key, "http://") || strings.HasPrefix(key, "https://") {
+			return auth, registry, errors.New("credentials key has https[s]:// prefix")
+		}
+
+		// TODO: check for tag and digest in key
```

Comment:
Should we do that?

Reply:
Yes, and reject that, I think.

We can always, with little effort, allow things later; once input is accepted, starting to reject it is much more difficult.

Reply:
Done :+1: